packages/babel-plugin/src/babel-plugin.ts

```diff
@@ -23,17 +24,42 @@ import {
   isCompiledStyledTaggedTemplateExpression,
   isCompiledCSSMapCallExpression,
 } from './utils/is-compiled';
+import { isTransformedJsxFunction } from './utils/is-jsx-function';
 import { normalizePropsUsage } from './utils/normalize-props-usage';
 import { visitXcssPropPath } from './xcss-prop';
 
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const packageJson = require('../package.json');
 const JSX_SOURCE_ANNOTATION_REGEX = /\*?\s*@jsxImportSource\s+([^\s]+)/;
-const JSX_ANNOTATION_REGEX = /\*?\s*@jsx\s+([^\s]+)/;
 const DEFAULT_IMPORT_SOURCE = '@compiled/react';
 
 let globalCache: Cache | undefined;
 
+const FindClassicJsxPragmaImport: Visitor<State> = {
+  ImportSpecifier(path, state) {
+    const specifier = path.node;
+
+    t.assertImportDeclaration(path.parent);
```

Comment:
Does this throw if it isn't an import declaration?

Reply:
<img width="1183" alt="Screenshot 2023-12-29 at 14 44 38" src="https://github.com/atlassian-labs/compiled/assets/2908767/b83b586b-a70b-4813-8bc6-e9c50c47ce8c">

yeah, provides both type checking and a runtime check